### ISSUE TITLE
Moving left on a slope teleports the player

### ISSUE THREAD

**Issue by svercl** [RandyGaul/player2d#7]
![As you can see here](https://user-images.githubusercontent.com/3598852/54035439-fa262780-417e-11e9-9b51-26555d37295f.gif)

**Comment by RandyGaul** [comment]
It’s a bug with collision normal flipping in some cases. I just need to do an audit and make sure normal sign is always consistent.

**Comment by RandyGaul** [comment]
I got a 100% repro using the frame-stepping code. Turns out to be an order of operations bug in the player stepping logic in main.cpp. There was a case where the sweep would return 1.0f, signaling freedom to move forward, and would exit the loop before applying any NGS. When running up a slope, this means the `skin_buffer` can be violated just as the player goes over an edge, since NGS was skipped.

There were also some minor bugs with c2CapsuletoPolyManifold, but had no real effect on the way the character performed.
